mobile/native/src/db/models.rs

```diff
@@ -551,11 +555,265 @@ impl From<crate::trade::order::FailureReason> for FailureReason {
     }
 }
 
+#[derive(Insertable, Debug, Clone, PartialEq)]
+#[diesel(table_name = payments)]
+pub(crate) struct PaymentInsertable {
+    #[diesel(sql_type = Text)]
+    pub payment_hash: String,
+    #[diesel(sql_type = Nullabel<Text>)]
+    pub preimage: Option<String>,
+    #[diesel(sql_type = Nullable<Text>)]
+    pub secret: Option<String>,
+    pub htlc_status: HtlcStatus,
+    #[diesel(sql_type = Nullable<BigInt>)]
+    pub amount_msat: Option<i64>,
+    pub flow: Flow,
+    pub created_at: i64,
+    pub updated_at: i64,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, FromSqlRow, AsExpression)]
+#[diesel(sql_type = Text)]
+pub enum HtlcStatus {
+    Pending,
+    Succeeded,
+    Failed,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, FromSqlRow, AsExpression)]
+#[diesel(sql_type = Text)]
+pub enum Flow {
+    Inbound,
+    Outbound,
+}
+
+impl PaymentInsertable {
+    pub fn insert(payment: PaymentInsertable, conn: &mut SqliteConnection) -> Result<()> {
+        let effected_rows = diesel::insert_into(payments::table)
+            .values(&payment)
+            .execute(conn)?;
+
+        ensure!(effected_rows > 0, "Could not insert order");
+
+        Ok(())
+    }
+
+    /// updates the status of the given order in the db
+    pub fn update(
+        payment_hash: String,
+        htlc_status: HtlcStatus,
+        preimage: Option<String>,
+        secret: Option<String>,
+        conn: &mut SqliteConnection,
+    ) -> Result<i64> {
+        let updated_at = OffsetDateTime::now_utc().unix_timestamp();
```

Comment:
I like the fact that we don't need to pass in this update time manually, and it got hidden deep :)